comps/lvms/tgi-llava/lvm_tgi.py

```diff
@@ -87,7 +88,14 @@ async def lvm(request: Union[LVMDoc, LVMSearchedMultimodalDoc]) -> Union[TextDoc
         top_k = request.top_k
         top_p = request.top_p
 
-    image = f"data:image/png;base64,{img_b64_str}"
+    if img_b64_str:
+        image = f"data:image/png;base64,{img_b64_str}"
+    else:
+        # Work around an issue where LLaVA-NeXT is not providing good responses when prompted without an image.
+        # Provide an image and then instruct the model to ignore the image.
+        image = "https://raw.githubusercontent.com/opea-project/GenAIExamples/refs/tags/v1.0/AudioQnA/ui/svelte/src/lib/assets/icons/png/audio1.png"
```

Comment:
Is that this issue we are facing with LLaVA-NeXT?
https://github.com/huggingface/transformers/issues/29835

Reply:
No, I'm not getting an error, it's just giving bad results. For example, this is prompt to tgi-gaudi with LLaVA-NeXT:
```
curl http://${host_ip}:${LLAVA_SERVER_PORT}/generate \
    -X POST \
    -d '{"inputs":"What is deep learning?","parameters":{"max_new_tokens":128, "seed": 42}}' \
    -H 'Content-Type: application/json'
{"generated_text":"\n\n\n\n\n\n\n"}
```
I was also able to reproduce those same results with just using transformers (without tgi-gaudi), which is why I am thinking it's an issue with LLaVA NeXT. LLaVA 1.5 is working fine (which is what is being used for Xeon).

Here is an example of the same question asked using my workaround to give an image and ask in the prompt to ignore the image:
```
curl http://${host_ip}:${LLAVA_SERVER_PORT}/generate \
    -X POST \
    -d '{"inputs":"![](https://raw.githubusercontent.com/opea-project/GenAIExamples/refs/tags/v1.0/AudioQnA/ui/svelte/src/lib/assets/icons/png/audio1.png)USER: Ignore the image and just answer the question. What is deep learning? ASSISTANT:","parameters":{"max_new_tokens":256, "seed": 42}}' \
    -H 'Content-Type: application/json'
{"generated_text":"Deep learning is a subset of machine learning that involves the use of neural networks to model and solve complex problems. It is a key part of artificial intelligence and is used in a wide range of applications, including image and speech recognition, decision-making, and natural language processing."}
```